File: frontend/src/types/trace.ts
// 单条 Agent 执行事件。
export interface TraceEventInfo {
  id: string
  step: number
  agent_name: string
  status: string
  message: string
  input_preview: string
  output_preview: string
  created_at: string
}

// 单次轨迹详情。
export interface TraceInfo {
  id: string
  task_type: string
  status: string
  input_summary: string
  output_summary: string
  created_at: string
  completed_at?: string | null
  events: TraceEventInfo[]
}

// 轨迹列表项。
export interface TraceListItem {
  id: string
  task_type: string
  status: string
  input_summary: string
  output_summary: string
  created_at: string
}

// 轨迹列表响应。
export interface TraceListResponse {
  items: TraceListItem[]
  total: number
}

File: backend/app/schemas/trace.py
"""Agent 执行轨迹接口结构。"""

from pydantic import BaseModel


class TraceEventInfo(BaseModel):
    """单步轨迹事件。"""

    id: str
    step: int
    agent_name: str
    status: str
    message: str
    input_preview: str = ""
    output_preview: str = ""
    created_at: str


class TraceInfo(BaseModel):
    """单次轨迹总览。"""

    id: str
    task_type: str
    status: str
    input_summary: str = ""
    output_summary: str = ""
    created_at: str
    completed_at: str | None = None
    duration_ms: int | None = None
    events: list[TraceEventInfo] = []


class TraceListItem(BaseModel):
    """轨迹列表项，用于前端时间线概览。"""

    id: str
    task_type: str
    status: str
    input_summary: str = ""
    output_summary: str = ""
    created_at: str
    duration_ms: int | None = None


class TraceListResponse(BaseModel):
    """轨迹列表响应。"""

    items: list[TraceListItem]
    total: int
